src/internal.c

```diff
@@ -4195,8 +4235,15 @@ static int DoKexDhReply(WOLFSSH* ssh, byte* buf, word32 len, word32* idx)
         }
     }
 
-    if (ret == WS_SUCCESS)
-        ret = GenerateKeys(ssh, enmhashId);
+    if (ret == WS_SUCCESS) {
+        ret = GenerateKeys(ssh, enmhashId,
+#ifndef WOLFSSH_NO_ECDH_NISTP256_KYBER_LEVEL1_SHA256
+                           !ssh->handshake->useEccKyber
```

Comment:
Can we make this more generic? Something along the lines of useKeyPadding. If we expand supported post quantum algorithms in the future than we are not adding more ifdef's and specific flag checks here.

Reply:
It would be helpful too with removing the magic number '1' in the else case of this macro guard.